core/src/main/scala/com/gu/cm/S3ConfigurationSource.scala

```diff
@@ -9,10 +9,11 @@ import com.typesafe.config.{Config, ConfigFactory}
 
 import scala.util.{Failure, Success, Try}
 
-class S3ConfigurationSource(s3: AmazonS3Client, identity: Identity, bucket: String) extends ConfigurationSource {
+class S3ConfigurationSource(s3: AmazonS3Client, identity: Identity, bucket: String, version: Option[Int]) extends ConfigurationSource {
 
   override def load: Config = {
-    val configPath = s"config/${identity.region}-${identity.stack}.conf"
+    val fileVersion = ".v" + version.getOrElse("")
```

Comment:
I think you want `version.map(".v" + _).getOrElse("")` (or use `fold`)
Otherwise fileversion would be `.v` when `version = None`